libevmasm/AssemblyItem.cpp

```diff
@@ -195,14 +201,18 @@ size_t AssemblyItem::bytesRequired(size_t _addressLength, langutil::EVMVersion _
 
 size_t AssemblyItem::arguments() const
 {
-	if (hasInstruction())
+	if (hasInstruction() && type() != CallF && type() != JumpF && type() != RetF)
```

Comment:
It would be more robust to handle these three item types first, before we check for the presence of an instruction. Then here assert against the corresponding instructions (rather than items):

```suggestion
	if (type() == CallF || type() == JumpF)
		return functionSignature().argsNum;
	else if (type() == RetF)
		return static_cast<size_t>(data());
	else if (hasInstruction())
	{
		solAssert(
			instruction() != Instruction::CALLF &&
			instruction() != Instruction::JUMPF &&
			instruction() != Instruction::RETF
		);
```

This would detect mistakes such as someone creating `Operation` containing one of those instructions rather than using the right item type. Currently we'd fall into the final `else` and return 0 instead.

Reply:
Right. Done